server/plugins/LFUGens.cpp

```diff
@@ -2785,6 +2788,7 @@ static inline bool check_gate_ar(EnvGen * unit, int i, float & prevGate, float *
 
 static inline bool EnvGen_nextSegment(EnvGen * unit, int & counter, double & level)
 {
+    if (unit->m_stage == ENVGEN_NOT_STARTED) { return true; }
```

Comment:
This introduced a bug where EnvGen.ar would not work:

```supercollider
// EnvGen.ar should continue to work.
// this should play the SinOsc
{ SinOsc.ar(440, 0, EnvGen.ar(Env([0, 1, 0],[0, 0.5], -3), Impulse.ar(1))) }.play
```

and was reverted for the 3.7.2 release https://github.com/supercollider/supercollider/commit/0f698556cabac7884936208edbc100ec26926a88

The other minor refactoring in this PR did make it in and are fine.